cmake/compiler/xcc/generic.cmake

```diff
@@ -6,12 +6,13 @@ find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}${CC}   PATHS ${TOOLCHAIN_HOME} NO
 find_program(CMAKE_GCOV ${CROSS_COMPILE}gcov   PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
 
 if(CMAKE_C_COMPILER STREQUAL CMAKE_C_COMPILER-NOTFOUND)
-  message(FATAL_ERROR "Zephyr was unable to find the toolchain. Is the environment misconfigured?
+  message(FATAL_ERROR "Zephyr was unable to find ${CROSS_COMPILE}${CC}. Is the environment misconfigured?
```

Comment:
FWIW, this particular error often shows up not because the tool wasn't found, but because running it with no arguments produces an error return.  The cmake behavior, setting the -NOTFOUND variable, is the same in both cases.

That error tends often to occur (on **some** versions of the Cadence tools) when XTENSA_CORE is set to an invalid value (vs. when it's not set, when the tool exits with success and produces an error message enumerating installed cores, sigh).

FWIW: my setup for doing Cadence builds manually is to scrub everything from the environment except XTENSA_TOOLS_ROOT and XTENSA_CORE and put the relevant toolchain's bin directory at the head of PATH.  This works everywhere AFAICT, but I don't have a recipe for getting SOF/Zephyr to do that; things are pretty thick.

Reply:
Thanks for the feedback, I will test more and adjust the message.

> but I don't have a recipe for getting SOF/Zephyr to do that;

I have some vaguely related xtensa-build-zephyr.py PR coming soon, stay tuned...

Reply:
I tested more and I could not reproduce what you described so I'm not going to touch the error message in this PR (which solves a different problem). Sorry.

I tried `find_program( ${CROSS_COMPILE}fail.sh ...)` where `xt-fail.sh` is just `exit 1` and I get a different and good error message:
```
  Executing the below command failed.  Are permissions set correctly?

    XCC/install/tools/RI-2022.10-linux/XtensaTools/bin/xt-fail.sh --version
```